oauth_generic.go

```diff
@@ -104,12 +108,28 @@ func (c genericOauthClient) inspectOauthAccessToken(ctx context.Context, accessT
 		return nil, errors.New("unable to inspect access token")
 	}
 
-	var inspectResponse InspectResponse
-	if err := limitedJsonUnmarshal(resp.Body, infoRequestMaxLen, &inspectResponse); err != nil {
+	// since we don't know what the JSON from the server will look like, we create a
+	// generic interface and then map manually to values set in the config
+	var genericInterface interface{}
+	if err := limitedJsonUnmarshal(resp.Body, infoRequestMaxLen, &genericInterface); err != nil {
 		return nil, err
 	}
-	if inspectResponse.Error != "" {
-		return nil, errors.New(inspectResponse.Error)
+
+	m := genericInterface.(map[string]interface{})
+
+	// map each relevant field in inspectResponse to the mapped field from the config
+	var inspectResponse InspectResponse
+	if (m[c.MapUserID] != nil) {
+	  inspectResponse.UserID = m[c.MapUserID].(string)
```

Comment:
Currently, this will crash if the value isn't a `string`. Instead, you'll want to also check that the type coercion worked by checking for a second return value:

```go
var ok bool
inspectResponse.UserID, ok = m[c.MapUserID].(string)
```

However, since we only want to avoid crashes and fall back to the default value if the type coercion fails, you can just ignore the second value:

```go
inspectResponse.UserID, _ = m[c.MapUserID].(string)
```

([Some demo code with more info about this](https://play.golang.org/p/7tZe4Gpc0ZX))

Reply:
Done in b262fa144c995d206c02aa4ed378c9d0b234c77a